slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -54,23 +56,57 @@ class SLAMServiceImpl final : public SLAMService::Service {
     // started & executed.
     void CreateMap();
 
+    // GetActionMode returns the slam action mode from the provided
+    // parameters.
+    SLAMServiceActionMode GetActionMode();
+
+    // SetUpMapBuilder loads the lua file with default cartographer config
+    // parameters depending on the action mode.
+    void SetUpMapBuilder();
+
+    // OverwriteMapBuilderParameters overwrites cartographer specific
+    // MapBuilder parameters.
+    void OverwriteMapBuilderParameters();
+
     std::string path_to_data;
     std::string path_to_map;
+    std::string configuration_directory;
     std::string config_params;
     std::string port;
     std::string camera_name;
     std::chrono::milliseconds data_rate_ms;
     std::chrono::seconds map_rate_sec;
     std::string slam_mode;
     std::atomic<bool> offlineFlag{false};
+    mapping::MapBuilder mapBuilder;
+
+    // -- Cartographer specific config params:
+    // MAP_BUILDER.pose_graph
+    int optimize_every_n_nodes = 3;
+    // TRAJECTORY_BUILDER.trajectory_builder_2d.submaps
+    int num_range_data = 100;
+    // TRAJECTORY_BUILDER.trajectory_builder_2d
+    float missing_data_ray_length = 25.0;
+    float max_range = 25.0;
+    float min_range = 0.2;
+    // TRAJECTORY_BUILDER.pure_localization_trimmer
+    int max_submaps_to_keep = 3;  // LOCALIZATION only
+    // MAP_BUILDER.pose_graph.overlapping_submaps_trimmer_2d
+    int fresh_submaps_count = 3;      // UPDATING only
+    double min_covered_area = 1.0;    // UPDATING only
+    int min_added_submaps_count = 1;  // UPDATING only
+    // MAP_BUILDER.pose_graph.constraint_builder.ceres_scan_matcher
+    double occupied_space_weight = 20.0;
+    double translation_weight = 10.0;
+    double rotation_weight = 1.0;
 
    private:
     int starting_scan_number = 0;
     int picture_print_interval = 50;
-    std::string configuration_directory = "../lua_files";
-    std::string configuration_mapping_basename = "mapping_new_map.lua";
-    std::string configuration_localization_basename = "locating_in_map.lua";
-    std::string configuration_update_basename = "updating_a_map.lua";
+    const std::string configuration_mapping_basename = "mapping_new_map.lua";
+    const std::string configuration_localization_basename =
+        "locating_in_map.lua";
+    const std::string configuration_update_basename = "updating_a_map.lua";
```

Comment:
Where can I find this files for testing locally?

Reply:
They're in the `lua_files` directory within `viam-cartographer`

Reply:
thanks!